lib/config/generator.js

```diff
@@ -0,0 +1,222 @@
+var utils = require('../utils');
+
+prompt.message = '';
+prompt.delimiter = '';
+prompt.start();
+
+/**
+ * Script that walks the user through the autoconfig flow
+ * @type {String[]}
+ */
+var prompts = [
+    {
+        name: 'Please choose a preset number:'.green,
+        require: true,
+        pattern: /[0-9]*/
+    },
+    {
+        name: 'Create an (e)xception for this rule, or (f)ix the errors yourself?'.green,
+        require: true,
+        pattern: /(e|f)/
+    }
+];
+
+function Generator() {
+    this._config = {};
+}
+
+Generator.prototype = {
+    /**
+     * Generates a configuration object based for the given path
+     * based on the best fitting preset
+     *
+     * @param  {String} path - The path containing file(s) used to guide the configuration
+     *
+     * @return {Promise} Resolved with the generated, JSCS configuration
+     */
+    generate: function(path) {
+        var checker = getChecker();
+        var _path = normalizePath(path, checker.getBasePath());
+        var presetNames = Object.keys(checker.getPresets());
+        var _this = this;
+        var errorsCollection;
+
+        console.log('Checking', _path, 'against the presets');
+
+        return Vow
+        .all(presetNames.map(this._checkAgainstPreset.bind(this, _path)))
+        .then(function generateErrorsCollection(results) {
+            errorsCollection = [].concat.apply([], results);
+            return errorsCollection;
+        })
+        .then(this._showErrorCounts.bind(this, presetNames))
+        .then(this._getUserPresetChoice.bind(this, prompts[0]))
+        .then(function getPresetIndexFromPrompt(choiceObj) {
+            return choiceObj[prompts[0].name] - 1;
+        })
+        .then(function storePresetConfiguration(presetIndex) {
+            var presetName = presetNames[presetIndex];
+            console.log('You chose the ' + presetName + ' preset');
+            this._config.preset = presetName;
+            return presetIndex;
+        }.bind(this))
+        .then(function getErrorListForChosenPreset(presetIndex) {
+            return errorsCollection[presetIndex].getErrorList();
+        })
+        .then(function showViolatedRules(errorList) {
+            var violatedRuleCount = getUniqueErrorNames(errorList).length;
+
+            if (!violatedRuleCount) { return this._config; }
+
+            console.log(_path + ' violates ' + violatedRuleCount + ' rule' + (violatedRuleCount > 1 ? 's' : ''));
+
+            var errorPrompts = generateRuleHandlingPrompts(errorList);
+
+            // TODO: Extract method?
+            return this._getUserViolationChoices(errorPrompts)
+            .then(function handleViolatedRules(choices) {
+                errorPrompts.forEach(function(errorPrompt, idx) {
+                    var associatedRuleName = errorList[idx].rule;
+                    var userChoice = choices[errorPrompt];
+
+                    _this._processViolationChoice(userChoice, associatedRuleName);
```

Comment:
You avoided `_this` everywhere, why not here? Just pass `this` as second arg to forEach.